pages/docs/concepts/asyncapi-document/reusing-traits.md

```diff
@@ -0,0 +1,105 @@
+---
+title: Reusing Traits
+weight: 200
+---
+
+Traits are a powerful way to define properties that can be reused across multiple message objects within the specification. Reusing traits promotes code maintainability, reduces duplication, and makes your AsyncAPI documents cleaner and easier to manage.
+
+## Defining Traits
+
+Traits are defined under the `components` section of your AsyncAPI document, within `operationTraits` or `messageTraits`, depending on whether you want to apply them to operations or messages, respectively. Each trait is given a unique name and contains the properties that will be applied.
+
+```mermaid
+graph TD
+    A[AsyncAPI Document] --> B{Components}
+    B --> C[operationTraits]
+    B --> D[messageTraits]
+    C --> E{Trait 1}
+    D --> G{Trait 2}
+    E --> I[Properties]
+    G --> K[Properties]
+```
+
+## Applying Traits to Operations
+
+Once a trait is defined, you can apply it to an operation using the `$ref` keyword in the `traits` section of the operation. The `$ref` value should point to the path of the trait within the `components` section.
+
+```mermaid
+graph LR
+    A((User Signup Operation))
+    B[traits]
+    C[$ref of kafka trait]
+    A-->B
+    B-->C
+```
+
+Here's applying the `kafka` trait to an operation:
+
+```yml
```

Comment:
is this a valid document? almost all on root, and kinda confusing that `$ref` is there with other props 🤔 

also if you show how to use trait, also show how the trait looks like

and most important, show how resulting document will look like after traits are really applied to document